tests/sender/send_data.py

```diff
@@ -315,3 +322,253 @@ def test_sender_as_handler_static(self):
+                                     chain=self.chain,
+                                     check_hostname=False,
+                                     verify_mode=CERT_NONE,
+                                     verify_config=False)
+
+        wrong_cert_message = "Error in the configuration, " + \
+                             wrong_cert.cert + \
+                             " is not a file or the path does not exist"
+
+        with self.assertRaises(DevoSenderException) as result:
+            wrong_cert.check_config_files_path()
+
+        self.assertEqual(wrong_cert_message, str(result.exception))
+
+    def test_config_files_path_incorrect_chain(self):
+        wrong_chain = SenderConfigSSL(address=(self.server, self.port),
+                                      key=self.chain,
+                                      cert=self.cert,
+                                      chain="Incorrect chain",
+                                      check_hostname=False,
+                                      verify_mode=CERT_NONE,
+                                      verify_config=False)
+        wrong_chain_message = "Error in the configuration, " + \
+                              wrong_chain.chain + \
+                              " is not a file or the path does not exist"
+
+        with self.assertRaises(DevoSenderException) as result:
+            wrong_chain.check_config_files_path()
+
+        self.assertEqual(wrong_chain_message, str(result.exception))
+
+    def test_config_cert_key_standard_case(self):
+        """
+        Test that verifies that a compatible certificate
+        and key are detected.
+        """
+
+        engine_config = SenderConfigSSL(address=(self.server, self.port),
+                                        key=self.key,
+                                        cert=self.cert,
+                                        chain=self.chain,
+                                        check_hostname=False,
+                                        verify_mode=CERT_NONE,
+                                        verify_config=False)
+        result = engine_config.check_config_certificate_key()
+        self.assertTrue(result)
+
+    def test_config_cert_key_incompatible_case(self):
+        """
+        Test that verifies that an incompatible
+        certificate with a key raises an exception.
+        """
+
+        engine_config = SenderConfigSSL(
+            address=(self.server, self.port),
+            key=self.key,
+            cert="{!s}/local_certs/keys/server/server_cert.pem".
+            format(os.path.dirname(os.path.abspath(__file__))),
+            chain=self.chain,
+            check_hostname=False,
+            verify_mode=CERT_NONE,
+            verify_config=False)
+
+        with self.assertRaises(DevoSenderException) as result:
+            engine_config.check_config_certificate_key()
+
+        expected_exceptions = "Error in the configuration, the key: " + \
```

Comment:
**nitpick(non-blocking):** This data type is simple, yet it is defined with a plural name. I believe you should leave it as singular.